### ISSUE TITLE
Compilation error

### ISSUE THREAD

**Issue by ThibaultMetais** [eddelbuettel/rquantlib#186]
When installing RQuantLib in R I get this error:

```
vanilla.cpp: In function ‘Rcpp::List europeanOptionEngine(std::string, double, double, double, double, double, double, Rcpp::Nullable<Rcpp::Vector<14, Rcpp::PreserveStorage> >, Rcpp::Nullable<Rcpp::Vector<14, Rcpp::PreserveStorage> >)’:
vanilla.cpp:89:9: error: ‘QL_DEPRECATED_DISABLE_WARNING’ was not declared in this scope
   89 |         QL_DEPRECATED_DISABLE_WARNING
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
vanilla.cpp:94:9: error: ‘QL_DEPRECATED_ENABLE_WARNING’ was not declared in this scope
   94 |         QL_DEPRECATED_ENABLE_WARNING
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
vanilla.cpp: In function ‘Rcpp::List americanOptionEngine(std::string, double, double, double, double, double, double, int, int, std::string, Rcpp::Nullable<Rcpp::Vector<14, Rcpp::PreserveStorage> >, Rcpp::Nullable<Rcpp::Vector<14, Rcpp::PreserveStorage> >)’:
vanilla.cpp:188:9: error: ‘QL_DEPRECATED_DISABLE_WARNING’ was not declared in this scope
  188 |         QL_DEPRECATED_DISABLE_WARNING
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
vanilla.cpp:190:9: error: ‘QL_DEPRECATED_ENABLE_WARNING’ was not declared in this scope
  190 |         QL_DEPRECATED_ENABLE_WARNING
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
vanilla.cpp:200:13: error: ‘option’ was not declared in this scope; did you mean ‘optind’?
  200 |             option.setPricingEngine(fdcnengine);
      |             ^~~~~~
      |             optind
make: *** [/usr/lib/R/etc/Makeconf:177: vanilla.o] Error 1
make: *** Waiting for unfinished jobs....
ERROR: compilation failed for package ‘RQuantLib’
```

I have R 3.6.3, libquantlib0-dev (1.17-2build1) and trying to install RQuantlib 0.4.21.

The same problem happens when I try the same thing with R 4.2.1.

**Comment by eddelbuettel** [comment]
These are ancient versions:
- R 3.6.3 is from Feb 2020, try is maybe with the matching QuantLib and RQuantLib from 2020
- libquantlib0-dev 1.17 is ancient from Dec 2019, try it with a matching RQuantLib from 2019

You are probably looking at RQuantLib 0.4.10 or 0.4.11 then.  If you have are trying this because R and QuantLib are already on your Linux distro (Debian? Ubuntu?) then r-cran-rquantlib may too.

Otherwise, with _current_ or at least _recent_ versions everything works.  See CRAN at https://cloud.r-project.org/web/checks/check_results_RQuantLib.html

**Comment by ThibaultMetais** [comment]
Thank you, it indeed works with the version 0.4.11 !

Is there a reason why libquantlib0-dev more recent versions are not available via apt ?
It would be convenient since installing the package via the tar.gz file takes forever to compile.

**Comment by eddelbuettel** [comment]
Salut Thibault -- glad it works.

You may be unaware that distributions (such as Debian and Ubuntu) have 'releases' that are made in time.  And for a given releases its packages remain fixed.  You still have not told me which distro and release you are using but as RQuantLib author (and maintainer, as well as maintainer of QuantLib) I can assure you it _is_ in Debian and Ubuntu. Here are two links to where you can search for packages, I use (source) package QuantLib here you can also search for binary packages (ie r-cran-rquantlib)

- Debian showing QuantLib 1.15 to 1.33 available:  https://packages.debian.org/search?keywords=quantlib&searchon=sourcenames&suite=all&section=all
- Ubuntu showing QuantLib 1.17 to 1.33 available:  https://packages.ubuntu.com/search?keywords=quantlib&searchon=sourcenames&suite=all&section=all

So in short: if you want newer packages, run a newer distribution release.   I will close this as there is actual issue with RQuantLib.
